core/include/cubos/core/net/tcp_listener.hpp

```diff
@@ -0,0 +1,72 @@
+/// @file
+/// @brief Class @ref cubos::core::net::TcpListener.
+/// @ingroup core-net
+
+#pragma once
+
+#include <cubos/core/api.hpp>
+#include <cubos/core/net/address.hpp>
+#include <cubos/core/net/tcp_stream.hpp>
+
+#ifdef _WIN32
+#include <winsock2.h>
+#pragma comment(lib, "Ws2_32.lib")
+#else
+#include <sys/socket.h>
+// Socket type. Differs from Windows (SOCKET) and POSIX (int).
+#define SOCKET int
+// Represents an invalid socket.
+#define INVALID_SOCKET (-1)
+#endif
```

Comment:
Are you sure we need to include this in the .hpp? Read my similar comment in #1257.